patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -1505,6 +1505,21 @@
 +    }
 +
 +    /**
++     * <p>Gets the {@code PathNodeType} for this block when the block is in the 3*3 grid around the
++     * path node. Called in {@link net.minecraft.pathfinding.WalkNodeProcessor#checkNeighborBlocks(IBlockAccess, int, int, int, net.minecraft.pathfinding.PathNodeType)}
++     *
++     * <p>Returns {@code original} for vanilla behavior.
++     *
++     * <p>Implementations: be aware that the {@code pos} argument may be mutable.
++     *
++     * @return the new path node type, or the default value
++     */
++    public net.minecraft.pathfinding.PathNodeType getAiEnvironmentalPathNodeType(IBlockState state, IBlockAccess world, BlockPos pos, net.minecraft.pathfinding.PathNodeType original)
++    {
++        return isBurning(world, pos) ? net.minecraft.pathfinding.PathNodeType.DANGER_FIRE : original;
```

Comment:
@mezz Is the enum constant here supposed to be remapped? However, it is not remapped to srg names.

Reply:
Enum names actually have an internal use in Java, so remapping them without breaking code would not be possible

Reply:
Now I figured that it'd be fine. https://github.com/MinecraftForge/MinecraftForge/blob/16bfd8cef1d12ee9ca0de1122addaf9916767ae9/patches/minecraft/net/minecraft/pathfinding/WalkNodeProcessor.java.patch#L7

By the way, I corrected the node type for above patch, which should be a fire threat instead of a direct fire damage.

Reply:
iirc the decompiler can figure out the exact name of Enum constants, so they don't have actual SRG names

```
<williewillus> gf PathNodeType.DANGER_FIRE
<MCPBot_Reborn> === MC 1.12: net/minecraft/pathfinding/PathNodeType.DANGER_FIRE (bef.i) LOCKED ===
<MCPBot_Reborn> Name       : i => PathNodeType.DANGER_FIRE => DANGER_FIRE
<MCPBot_Reborn> Descriptor : Lbef; => Lnet/minecraft/pathfinding/PathNodeType;
<MCPBot_Reborn> Comment    : None
<MCPBot_Reborn> Last Change: 2017-06-13 12:00:11.931410-04:00 (_bot_update_)
```